client/landing/stepper/declarative-flow/newsletter.ts

```diff
@@ -157,9 +170,8 @@ const newsletter: Flow = {
 			switch ( _currentStep ) {
 				case 'launchpad':
 					return window.location.assign( `/view/${ siteSlug }` );
-
 				default:
-					return navigate( 'intro' );
+					return navigate( isComingFromMarketingPage ? 'newsletterSetup' : 'intro' );
```

Comment:
I liked having this conditional in the same line where it's used, instead of declaring on top of the file, and using down in the file.